contentcuration/contentcuration/frontend/shared/views/Icon.vue

```diff
@@ -1,45 +1,37 @@
-<script>
+<template>
 
-  import isArray from 'lodash/isArray';
-  import isObject from 'lodash/isObject';
-  import { VIcon } from 'vuetify/lib/components/VIcon';
+  <KIcon
+    v-bind="$attrs"
+    :icon="icon"
+    :color="color"
+    :class="{ 'rtl-flip': rtlFlip }"
+    v-on="$listeners"
+  />
 
-  const NO_TRANSLATE = 'notranslate';
+</template>
 
-  // Follows example of https://github.com/vuetifyjs/vuetify/blob/v1.5.22/packages/vuetify/src/components/VIcon/VIcon.ts#L142
-  export default {
-    name: 'Icon',
-    functional: true,
-
-    render(createElement, { data, children }) {
-      let classNames = data['class'];
+<script>
 
-      // Add `notranslate` class
-      if (isArray(classNames)) {
-        classNames.push(NO_TRANSLATE);
-      } else if (isObject(classNames)) {
-        classNames[NO_TRANSLATE] = true;
-      } else {
-        classNames = `${classNames || ''} ${NO_TRANSLATE}`;
-      }
+  import { themeTokens } from 'kolibri-design-system/lib/styles/theme.js';
 
-      data['class'] = classNames;
-      return createElement(
-        VIcon,
-        data,
-        data.props && data.props.iconName ? [data.props.iconName] : children
-      );
+  export default {
+    name: 'Icon',
+    props: {
+      icon: {
+        type: String,
+        required: true,
+      },
+
+      color: {
+        type: String,
+        required: false,
+        default: themeTokens.text,
```

Comment:
Removing this default slot, displays the error that default needs a value mandatorily. So is there any other default value that will be more suited here?

Reply:
Where is the error coming from exactly? I don't see how `Icon` component could give rise to the validation error since it is not `required`.

Reply:
It's an ESLint-related error.
![image](https://github.com/learningequality/studio/assets/95405559/b66e9a10-8634-4f1e-af23-eae69854cfef)

Reply:
@BabyElias I see, thank you. This just reminds folks to be explicit, but it doesn't mean that the default value can't be "empty". Could you set `default: null` rather than adding `eslint-disable`? I think it should help and it's a common way in our codebases:
![Screenshot from 2024-04-17 07-39-52](https://github.com/learningequality/studio/assets/13509191/88c7f9fd-9aa4-4750-80a0-8f3df43cfdb0)

Reply:
Ahh, Thanks! I got a little confused between the `null` and `empty` values. This makes much more sense. Have made the required changes accordingly.